syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.Aeropuerto";
option java_outer_classname = "AeropuertoProto";
option objc_class_prefix = "HLW";

package Aeropuerto;

service Serv_Aeropuerto{

//solicitud de despegue
  rpc enviar_despegue (Pista_Des) returns (Resp_Des){}

//asignacion pista de aterrizaje
  rpc enviar_aterrizaje (Pista_At) returns (Resp_At){}

//Enviar info pantallas
  rpc enviar_info (Id_Pantalla) returns (Resp_Pantalla){}


}

message Pista_Des{

  string Id_avion = 1;
  string Name_destino = 2;

}

message Resp_Des{
  bool Autorizacion_despegue = 1;
  string Pista_despegue = 2;
  int32 Altura_despegue = 3;
  int32 Posicion_despegue = 4;
  string Anterior_despegue = 5;
  string Ip_destino = 6;
  string Id_avion = 7;
  }


message Pista_At{
  string Id_avion = 1;

}

message Resp_At{

  bool Autorizacion_aterrizaje = 1;
  int32 Pista_aterrizaje = 2;
  int32 Altura_aterrizaje = 3;
  int32 Posicion_aterrizaje = 4;
  string Anterior_aterrizaje = 5;
  string Id_avion = 6;
}



message Id_Pantalla{
  string Nombre_Pantalla=1;
}

message Resp_Pantalla{
  string Salida1 = 1;
  string A_Donde1 = 2;
  string Pista_Salida1 = 3;
  string Hora_Salida1 = 4;
  string Salida2 = 5;
  string A_Donde2 = 6;
  string Pista_Salida2 = 7;
  string Hora_Salida2 = 8;

  string Llegda1 = 9;
  string De_donde1 = 10;
  string Pista_Llegada1 = 11;
  string Hora_Llegada1 = 12;
  string Llegda2 = 13;
  string De_donde2 = 14;
  string Pista_Llegada2 = 15;
  string Hora_Llegada2 = 16;
}
